### ISSUE TITLE
[Exception][CLion 2022.3] In file: file://C:/Users/dklimov/CLionProjects/mediasoup/CMakeLists.txt

### ISSUE THREAD

**Issue by dmitry-klimov** [ArtsiomCh/CMake#41]
# User Description
_User did not enter any detailed information._

# Stack Trace
```
com.intellij.psi.PsiInvalidElementAccessException: Element: class com.jetbrains.cmake.psi.CMakeCommandImpl #CMake  because: parent is null
invalidated at: see attachment
	at com.intellij.extapi.psi.ASTDelegatePsiElement.getManager(ASTDelegatePsiElement.java:76)
	at com.intellij.extapi.psi.ASTDelegatePsiElement.getManager(ASTDelegatePsiElement.java:53)
	at com.intellij.psi.impl.PsiElementBase.getProject(PsiElementBase.java:213)
	at com.cmakeplugin.utils.CMakePSITreeSearch.lambda$cleanCachesForDisposedProjects$5(CMakePSITreeSearch.java:69)
	at java.base/java.util.Collection.removeIf(Collection.java:576)
	at com.cmakeplugin.utils.CMakePSITreeSearch.lambda$cleanCachesForDisposedProjects$7(CMakePSITreeSearch.java:69)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.concurrent.ConcurrentHashMap$KeySpliterator.forEachRemaining(ConcurrentHashMap.java:3573)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at com.cmakeplugin.utils.CMakePSITreeSearch.cleanCachesForDisposedProjects(CMakePSITreeSearch.java:64)
	at com.cmakeplugin.utils.CMakePSITreeSearch.addFileListener(CMakePSITreeSearch.java:57)
	at com.cmakeplugin.utils.CMakePSITreeSearch.getCmakeFiles(CMakePSITreeSearch.java:92)
	at com.cmakeplugin.utils.CMakePSITreeSearch.existCommandDefFor(CMakePSITreeSearch.java:283)
	at com.cmakeplugin.utils.CMakePSITreeSearch.existFunctionDefFor(CMakePSITreeSearch.java:263)
	at com.cmakeplugin.annotator.CMakeAnnotatorUtils.annotateCommand(CMakeAnnotatorUtils.java:187)
	at com.cmakeplugin.annotator.CMakeAnnotator.annotateCLion(CMakeAnnotator.java:74)
	at com.cmakeplugin.annotator.CMakeAnnotator.annotate(CMakeAnnotator.java:23)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:131)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:108)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:344)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$7(GeneralHighlightingPass.java:277)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:304)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$8(GeneralHighlightingPass.java:307)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:93)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:307)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:274)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:222)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:97)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:57)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:405)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1086)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:397)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:666)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:622)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:590)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:396)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:372)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:182)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:370)
	at com.intellij.concurrency.JobLauncher

<TRIMMED STACKTRACE>
```

# Other Information
```
=== About ===
Build version: CLion 2022.3 Build: #CL-223.7571.171 November 28, 2022
Theme: Darcula
JRE: 17.0.5+1-b653.14, JetBrains s.r.o.
JVM: 17.0.5+1-b653.14, OpenJDK 64-Bit Server VM, JetBrains s.r.o.
Operating System: Windows 10 10.0 (amd64)

=== Plugins ===
Custom plugins: [Rust (0.4.183.5026-223), Meson Syntax Highlighter (203.1.0), CMake simple highlighter (223.2.0), CMD Support (1.0.5), Conan (1.2.0), File Watchers (223.7571.113), Batch Scripts Support (1.0.13), Ideolog (203.0.30.0)]
Disabled plugins:[Task Management (223.7571.171), Terminal (223.7571.171), Node.js Remote Interpreter (223.7571.171), New Executable C/C++ (1.3), JavaScript Intention Power Pack (223.7571.171), TSLint (223.7571.171), C/C++ Single File Execution (1.4), Styled Components & Styled JSX (223.7571.171), Less (223.7571.171), Mercurial (223.7571.171), GitHub (223.7571.171), Subversion (223.7571.171), HTML Tools (223.7571.171), Terminal for Gateway (223.7571.171), HTTP Client (223.7571.171), SSH Remote Run (223.7571.171), Lua (1.0.119), XPathView + XSLT (223.7571.171), Time Tracking (223.7571.171), Sass (223.7571.171), Perforce Helix Core (223.7571.171)]
```

**Comment by ArtsiomCh** [comment]
Can't reproduce, but can see through Stacktrace where it comes from.
Will be fixed in next release (within this week).
Thank you for reporting it!

**Comment by ArtsiomCh** [comment]
Fixed at version 223.3.0
